sherpa/astro/data.py

```diff
@@ -2788,6 +2788,27 @@ def set_background(self, bkg, id=None):
         external programs, such as XSPEC.
 
         """
+
+        if not isinstance(bkg, DataPHA):
+            raise ArgumentTypeErr("badarg", "bkg", "a PHA data set")
+
+        # Check that the background matches the source (i.e. self)
+        # dataset.  For this use case we require that the source has
+        # set up it's channel array (ie self.channel can not be None),
+        # as allowing the background dataset to have channels but not
+        # the source makes tracking the state harder than the ability
+        # is likely worth).
+        #
+        if self.channel is None:
+            raise DataErr("The channel field must be set before adding a background")
```

Comment:
```suggestion
            raise DataErr("The channel field must be set before adding a background.")
```

Reply:
Most (too lazy to check the numbers) of our existing errors do not end in a full stop.